packages/fuel-gauge/src/auth-testing.test.ts

```diff
@@ -6,11 +6,12 @@ const setupContract = getSetupContract('auth_testing_contract');
 
 let contractInstance: Contract;
 
-beforeAll(async () => {
-  contractInstance = await setupContract();
-});
+// TODO: Fix and unskip test
+describe.skip('Auth Testing', () => {
```

Comment:
Happy to take ownership.

However, I suspect we may have a misunderstanding here.

I will try to outline my rationale in detail first.

Please, see if it makes sense to you.

> _The tests not mentioned below are **passing**_.

# Test 1

This version https://github.com/FuelLabs/fuels-ts/commit/bb6709d1ebaf95bd3c31a2312a0bba48a0a20eb4 seems to work because the test always `throw new Error(...)` and always passes.

When I try it in this branch, it also passes.

> _The error from the `.get()` call is never asserted, the method may not be throwing._

```ts
  it.only('can check_msg_sender [with correct id, using get]', async () => {
    const wallet = getWallet();
    try {
      await contractInstance.functions.check_msg_sender({ value: wallet.address.toB256() }).get();

      throw new Error('it should have thrown');
    } catch (error) {
      expect(error).toBeTruthy();
    }
  });
```
When [corrected](https://github.com/FuelLabs/fuels-ts/pull/664#discussion_r1048328459), it became this:

```ts
  it.only('can check_msg_sender [with correct id, using get]', async () => {
    const wallet = getWallet();

    expect(async () => {
      await contractInstance.functions.check_msg_sender({ value: wallet.address.toB256() }).get();
    }).toThrow();
  });
```

This gives us this output, confirming that the method isn't throwing.

```console
$ pnpm test packages/fuel-gauge/src/auth-testing.test.ts

> fuels-ts@0.0.0 test /[...]/fuel/fuels-ts
> jest --runInBand "packages/fuel-gauge/src/auth-testing.test.ts"

 FAIL  packages/fuel-gauge/src/auth-testing.test.ts
  Auth Testing
    ✕ can check_msg_sender [with correct id, using get] (9 ms)
    ○ skipped can get is_caller_external
    ○ skipped can check_msg_sender [with correct id]
    ○ skipped can check_msg_sender [with incorrect id]

  ● Auth Testing › can check_msg_sender [with correct id, using get]

    expect(received).toThrow()

    Received function did not throw

      33 |     expect(async () => {
      34 |       await contractInstance.functions.check_msg_sender({ value: wallet.address.toB256() }).get();
    > 35 |     }).toThrow();
         |        ^
      36 |   });
      37 |
      38 |   it('can check_msg_sender [with incorrect id]', async () => {

      at Object.<anonymous> (packages/fuel-gauge/src/auth-testing.test.ts:35:8)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 3 skipped, 4 total
Snapshots:   0 total
Time:        0.976 s, estimated 1 s
Ran all test suites matching /packages\/fuel-gauge\/src\/auth-testing.test.ts/i.
/[...]/fuel/fuels-ts/packages/script/src/script.ts:72
  bytes: Uint8Array;
                  ^
ScriptResultDecoderError: Script returned non-zero result: 1

{
  "doc": "https://docs.rs/fuel-asm/latest/fuel_asm/enum.PanicReason.html",
  "reason": "unknown"
}

Reverts:
0xc2aeef19b63932439e6e4499580c12371e391fa274fa87f9640251da6d4accdb: 0 {"type":4,"val":"0x0","pc":"0x4454","is":"0x4230"}

Receipts:
[
  {
    "type": "Call",
    "from": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "to": "0xc2aeef19b63932439e6e4499580c12371e391fa274fa87f9640251da6d4accdb",
    "amount": "0x0",
    "assetId": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "gas": "0x5f56199",
    "param1": "0x5d9d062b",
    "param2": "0x3868",
    "pc": "0x4230",
    "is": "0x4230"
  },
  {
    "type": "Revert",
    "id": "0xc2aeef19b63932439e6e4499580c12371e391fa274fa87f9640251da6d4accdb",
    "val": "0x0",
    "pc": "0x4454",
    "is": "0x4230"
  },
  {
    "type": "ScriptResult",
    "result": "0x1",
    "gasUsed": "0xb660"
  }
]
    at Script.decodeCallResult (/[...]/fuel/fuels-ts/packages/script/src/script.ts:72:19)
    at InvocationCallResult.getDecodedValue (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/invocation-results.ts:23:61)
    at new InvocationResult (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/invocation-results.ts:18:27)
    at new InvocationCallResult (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/invocation-results.ts:65:9)
    at Function.build (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/invocation-results.ts:69:26)
    at FunctionInvocationScope.dryRun (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:186:72)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /[...]/fuel/fuels-ts/packages/fuel-gauge/src/auth-testing.test.ts:34:7 {
  logs: []
}
 ELIFECYCLE  Test failed. See above for more details.
```

# Test 2

Pretty much the same situation; if I run it like this, it works for the same reasons:

```ts
  it.only('can check_msg_sender [with incorrect id]', async () => {
    const wallet = getWallet();

    try {
      await contractInstance.functions
        .check_msg_sender({ value: wallet.address.toB256().replace('x', 'y') })
        .call();

      throw new Error('it should have thrown');
    } catch (error) {
      expect(error).toBeTruthy();
    }
  });
```

But this:

```ts
  it.only('can check_msg_sender [with incorrect id]', async () => {
    const wallet = getWallet();

    expect(async () => {
      await contractInstance.functions
        .check_msg_sender({ value: wallet.address.toB256().replace('x', 'y') })
        .call();
    }).toThrow();
  });
```

Will give us the following output:

```console
pnpm test packages/fuel-gauge/src/auth-testing.test.ts


> fuels-ts@0.0.0 test /[...]/fuel/fuels-ts
> jest --runInBand "packages/fuel-gauge/src/auth-testing.test.ts"

/[...]/fuel/fuels-ts/node_modules/.pnpm/@ethersproject+logger@5.7.0/node_modules/@ethersproject/logger/src.ts/index.ts:269
        const error: any = new Error(message);
                           ^
Error: Invalid b256 (argument="b256", value="0y233ce0d98f36e393a7e276c8ae06fd7c6d1fc97a1b5183c36beac37dfa645490", code=INVALID_ARGUMENT, version=0.0.0)
    at Logger.Object.<anonymous>.Logger.makeError (/[...]/fuel/fuels-ts/node_modules/.pnpm/@ethersproject+logger@5.7.0/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.Object.<anonymous>.Logger.throwError (/[...]/fuel/fuels-ts/node_modules/.pnpm/@ethersproject+logger@5.7.0/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at Logger.Object.<anonymous>.Logger.throwArgumentError (/[...]/fuel/fuels-ts/node_modules/.pnpm/@ethersproject+logger@5.7.0/node_modules/@ethersproject/logger/src.ts/index.ts:285:21)
    at B256Coder.throwError (/[...]/fuel/fuels-ts/packages/abi-coder/src/coders/abstract-coder.ts:13:16)
    at B256Coder.encode (/[...]/fuel/fuels-ts/packages/abi-coder/src/coders/b256.ts:19:18)
    at /[...]/fuel/fuels-ts/packages/abi-coder/src/coders/struct.ts:23:40
    at Array.map (<anonymous>)
    at StructCoder.encode (/[...]/fuel/fuels-ts/packages/abi-coder/src/coders/struct.ts:17:56)
    at /[...]/fuel/fuels-ts/packages/abi-coder/src/coders/tuple.ts:18:72
    at Array.map (<anonymous>)
    at TupleCoder.encode (/[...]/fuel/fuels-ts/packages/abi-coder/src/coders/tuple.ts:18:48)
    at AbiCoder.encode (/[...]/fuel/fuels-ts/packages/abi-coder/src/abi-coder.ts:126:31)
    at Interface.encodeFunctionData (/[...]/fuel/fuels-ts/packages/abi-coder/src/interface.ts:88:36)
    at createContractCall (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:12:37)
    at /[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:33:65
    at Array.map (<anonymous>)
    at FunctionInvocationScope.get calls [as calls] (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:33:46)
    at FunctionInvocationScope.updateScriptRequest (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:39:28)
    at FunctionInvocationScope.addCalls (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:72:14)
    at FunctionInvocationScope.addCall (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/base-invocation-scope.ts:67:14)
    at new FunctionInvocationScope (/[...]/fuel/fuels-ts/packages/contract/src/contracts/functions/invocation-scope.ts:12:15)
    at Object.<anonymous> (/[...]/fuel/fuels-ts/packages/contract/src/contracts/contract.ts:30:29)
    at /[...]/fuel/fuels-ts/packages/fuel-gauge/src/auth-testing.test.ts:43:10
    at Object.<anonymous> (/[...]/fuel/fuels-ts/node_modules/.pnpm/expect@29.3.1/node_modules/expect/build/toThrowMatchers.js:74:11)
    at Object.throwingMatcher [as toThrow] (/[...]/fuel/fuels-ts/node_modules/.pnpm/expect@29.3.1/node_modules/expect/build/index.js:309:21)
    at Object.<anonymous> (/[...]/fuel/fuels-ts/packages/fuel-gauge/src/auth-testing.test.ts:45:8)
    at Promise.then.completed (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/utils.js:289:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/utils.js:222:10)
    at _callCircusTest (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/run.js:248:40)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async _runTest (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/run.js:184:3)
    at async _runTestsForDescribeBlock (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/run.js:86:9)
    at async _runTestsForDescribeBlock (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/run.js:81:9)
    at async run (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/run.js:26:3)
    at async runAndTransformResultsToJestFormat (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:120:21)
    at async jestAdapter (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-circus@29.3.1/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
    at async runTestInternal (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-runner@29.3.1/node_modules/jest-runner/build/runTest.js:367:16)
    at async runTest (/[...]/fuel/fuels-ts/node_modules/.pnpm/jest-runner@29.3.1/node_modules/jest-runner/build/runTest.js:444:34) {
  reason: 'Invalid b256',
  code: 'INVALID_ARGUMENT',
  argument: 'b256',
  value: '0y233ce0d98f36e393a7e276c8ae06fd7c6d1fc97a1b5183c36beac37dfa645490'
}

 RUNS  packages/fuel-gauge/src/auth-testing.test.ts
 ELIFECYCLE  Test failed. See above for more details.
```